client/header/activity-panel/index.js

```diff
@@ -134,25 +135,45 @@ export class ActivityPanel extends Component {
 
 		// Don't show the inbox on the Home screen.
 		const { location } = this.props.getHistory();
-		const showInbox = isEmbedded || ! window.wcAdminFeatures.homescreen || location.pathname !== '/';
+		const showInbox =
+			isEmbedded ||
+			! window.wcAdminFeatures.homescreen ||
+			location.pathname !== '/';
 		const isPerformingSetupTask =
 			query.task &&
 			! query.path &&
 			( requestingTaskListOptions === true ||
-				(
-					taskListHidden === false &&
-					taskListComplete === false
-				)
-			);
+				( taskListHidden === false && taskListComplete === false ) );
 
-		return [
-			! isPerformingSetupTask && showInbox
-				? {
+		if ( ! taskListComplete && showInbox ) {
```

Comment:
generating the tab list was just increasing further in complexity and its already quite hard to reason about, so I maintain a second list of tabs for displaying setup. I think this `getTabs` method is ripe for refactoring (its not really easy to test it right now), but it was too much of a diversion to fix it here.

Reply:
I agree that this could be a good refactor target too. Given some of the new designs around this part of the app though, I think the current panels are going to go away, and possibly be replaced with more contextual panels... but even on that approach, this would need to be refactored to be more extensible too.

Reply:
There is some logic here that doesn't match. The criteria for showing the setup activity panel icon is `! taskListComplete`, but the task list also is hidden if `woocommerce_task_list_hidden` is set to "yes". In other words, it's possible to be in a state where the setup button is visible, but the task list isn't shown on the Homescreen once you've navigated there.

To reproduce, leave the task list incomplete and dismiss it via:

![Screen Shot 2020-07-29 at 1 38 15 PM](https://user-images.githubusercontent.com/1922453/88746958-e16a4780-d1a1-11ea-93aa-ad906a042338.png)

We could match the criteria, but I think it might be cool to leave it as is, but set the option `woocommerce_task_list_hidden` to "no" before navigating. This way, a user can dismiss the setup task list, but still access it.

Reply:
@psealock Thanks for the feedback, I've added a fix for this in https://github.com/woocommerce/woocommerce-admin/pull/4853/commits/0f243b2dd50e255879270e9e7039da156c08af84 Can you let me know if I'm on the right track? I tested it and it works as expected for me.